src/elementary/JoinedIdentifier/src/generate.hpp

```diff
@@ -0,0 +1,14 @@
+/**
+ *  @brief Private helper function to generate the id string.
+ */
+std::string generate( std::vector< std::string >&& strings ) const {
```

Comment:
Could this be done with `ranges::view::join`? That would be much easier, I think.

Reply:
When I wrote it, I was like "I wish I had ranges" :-)

Anyway, for the limited functionality offered by this library (for now), I don't think it merits adding a dependency - especially since the standard library meets our needs just fine.

Reply:
Oh yeah, I forgot you are not using ranges in this library.

Reply:
I would actually like to put PoPs in here as well, so we may pull in ranges at some point. But until then, I'm going through the effort of doing so.